### PR TITLE
feat(macOS): Support host DNS aliases for macos

### PR DESCRIPTION
**Pull request by Kern--**
Issue #, if available:
macOS portion of #1079 

*Description of changes:*
This change adds:
1) `host.finch.internal`
2) `host.docker.internal`

as DNS names that are aliased to the host's loopback address. This allows service and containers inside the VM to connect to services bound to the host's loopback device.

*Testing done:*
main.go
```
package main

import "net/http"

func main() {
        http.HandleFunc("/hello", func(w http.ResponseWriter, r *http.Request) {
                w.Write([]byte(`{"response": "ok"}`))
        })
        http.ListenAndServe(":8080", nil)
}
```

```
$ go run main.go

in another shell

$ limactl shell finch
$ curl host.finch.internal:8080/hello
{"response": "ok"}
$ curl host.docker.internal:8080/hello
{"response": "ok"}

```

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by pendo324 — DISMISSED** [review]

**Review by chews93319 — APPROVED** [review]

**Review by austinvazquez — COMMENTED** [review]
Can a e2e test be added to run a container and ping the DNS aliases here?

File: finch.yaml.d/mac.yaml
Comment (sondavidb) [inline finch.yaml.d/mac.yaml:46]: Do we want to have this alias? I understand having the Finch alias but this one seems like it could be a little confusing.

Comment (Pballer) [reply finch.yaml.d/mac.yaml:46]: Has the team discussed this yet?  Looks like the change is approved and only pending this final decision.

Comment (pendo324) [reply finch.yaml.d/mac.yaml:46]: Hey @Pballer, looks like @Kern-- just rebased and added an e2e test case for using the new domains. After the tests are passing, we should be good to merge.

**Review by austinvazquez — APPROVED** [review]

**Review by pendo324 — APPROVED** [review]

File: finch.yaml.d/mac.yaml
Comment (pendo324) [reply finch.yaml.d/mac.yaml:46]: Think we also need to resolve this comment before merging. @sondavidb are you ok with the "docker" alias for the sake of compatibility?

Comment (sondavidb) [reply finch.yaml.d/mac.yaml:46]: Yeah, it's fine, I'm not particularly opinionated on it. Thanks

**Pull request merged**
